src/priorityqueue.jl

```diff
@@ -351,13 +357,71 @@ function empty!(pq::PriorityQueue)
     pq
 end
 
+empty(pq::PriorityQueue) = PriorityQueue(empty(pq.xs), pq.o, empty(pq.index))
+
+function merge!(d::SortedDict{K, V, O1}, other::PriorityQueue{K, V, O}) where {K, V, O <: Ordering, O1 <: Ordering}
+    next = iterate(other, false)
+    while next !== nothing
+        (k, v), state = next
+        d[k] = v
+        next = iterate(other, state)
+    end
+    return d
+end
```

Comment:
Since the body of this function is identical to the body of the next function, we can invoke the second using `invoke`:

```julia
merge!(d::SortedDict{K,V,O1}, other::PriorityQueue{K,V, O}) where {K, V, O1, O} = invoke(merge!, (AbstractDict{K,V}, PriorityQueue{K,V,O}), d, other)
```

or more simply (after removing the type parameters on the next function)

```julia
merge!(d::SortedDict, other::PriorityQueue) = invoke(merge!, (AbstractDict, PriorityQueue), d, other)
```